Automattic/wp-calypso
# RTS: Details page

Fixes https://github.com/Automattic/dotcom-forge/issues/7812

## Proposed Changes

Creates a new Readymade Template details page within the Pattern Library.

![container-flamboyant-hofstadter calypso live_patterns_site-layouts_1](https://github.com/Automattic/wp-calypso/assets/1233880/a7dd3e28-d013-4a76-aadb-d66a796033ac)


A few notes:
- When going back to patterns, the page is automatically scrolled to the top. This is a known issue, ideally we should scroll to the RTS section again. It's not easy to solve because of [the current enforced scrolling behavior](https://github.com/Automattic/wp-calypso/blob/a900a7934a2be3ef5fe8d78a007e702b820dac38/client/my-sites/patterns/components/pattern-library/index.tsx#L130-L143), but I don't think it's a blocker for the MVP. We can address it later on.
- Current descriptions look empty. We should update them in the API to make them more exciting.
- The "Pick this layout" button requires changes from https://github.com/Automattic/wp-calypso/pull/92031

 

## Why are these changes being made?

To connect the PL with the new RT onboarding flow

## Testing Instructions

- Use the Calypso link below
- Go to `/patterns`
- Scroll down to the RTS section
- Click on a RT
- Make sure the details page is opened correctly
- Make sure it looks good in different viewports
- Make sure the "All about patterns" section at the bottom as a light color scheme
- Make sure the "Copy layout" button copies the template content to your clipboard (you can paste it in the site editor)
- Make sure the "Pick this layout" button triggers the RT onboarding flow implemented in https://github.com/Automattic/wp-calypso/pull/92031

## Pre-merge Checklist
